src/components/MultiHopTrade/hooks/useTradeExecution.tsx

```diff
@@ -0,0 +1,117 @@
+import { supportsETH, supportsEthSwitchChain } from '@shapeshiftoss/hdwallet-core'
+import { useCallback, useState } from 'react'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useErrorHandler } from 'hooks/useErrorToast/useErrorToast'
+import { usePoll } from 'hooks/usePoll/usePoll'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import type { Swapper2, TradeQuote2 } from 'lib/swapper/api'
+import { SwapperName } from 'lib/swapper/api'
+import { lifi } from 'lib/swapper/swappers/LifiSwapper/LifiSwapper2'
+import { thorchain } from 'lib/swapper/swappers/ThorchainSwapper/ThorchainSwapper2'
+import { assertUnreachable, isEvmChainAdapter } from 'lib/utils'
+import { selectPortfolioAccountMetadataByAccountId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { TRADE_POLL_INTERVAL_MILLISECONDS } from './constants'
+import { useAccountIds } from './useAccountIds'
+
+export const useTradeExecution = ({
+  swapperName,
+  tradeQuote,
+}: {
+  swapperName: SwapperName
+  tradeQuote: TradeQuote2
+}) => {
+  const [txId, setTxId] = useState<string | undefined>()
+  const [message, setMessage] = useState<string | undefined>()
+  const { poll } = usePoll()
+  const { showErrorToast } = useErrorHandler()
+  const wallet = useWallet().state.wallet
+
+  const { sellAssetAccountId } = useAccountIds({
+    sellAsset: tradeQuote.steps[0].sellAsset,
+  })
+
+  const sellAccountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, { accountId: sellAssetAccountId }),
```

Comment:
there is a `selectAccountNumberByAccountId` to make things easier but stylistic comment only

Reply:
`useTradeExecution` removed from this diff, to be tackled in a PR on top of this one in the stack

Reply:
@woodenfurniture FYI ^^